src/wrappers/OsipiBase.py

```diff
@@ -0,0 +1,210 @@
+        results = self.ivim_fit(*args, **kwargs)
+        
+        #self.parameter_estimates = self.ivim_fit(data, bvalues)
+        return results
+    
+    def osipi_print_requirements(self):
+        """
+        Prints the requirements of the algorithm.
+        Attributes that are currently checked for are:
+        required_bvalues (int):
+            The lowest number of b-values required.
+        required_thresholds (array-like)
+            1D array-like of two ints [least number of b-value thresholds, most number of b-value_thresholds].
+        required_bounds (bool):
+            Whether bounds are required or not.
+        required_bounds_optional (bool):
+            Whether bounds are optional or not
+        required_initial_guess (bool):
+            Whether an initial guess is required or not.
+        required_initial_guess_optional (bool):
+            Whether an initial guess is optional or not.
+        """
+        print("\n### Algorithm requirements ###")
+        
+        # Check if the attributes exist and print them
+        if hasattr(self, "required_bvalues"):
+            print(f"Number of b-values: {self.required_bvalues}")
+            
+        if hasattr(self, "required_thresholds"):
+            print(f"Numer of b-value thresholds [at least, at most]: {self.required_thresholds}")
+            
+        if hasattr(self, "required_bounds") and hasattr(self, "required_bounds_optional"):
+            if self.required_bounds:
+                print(f"Bounds required: {self.required_bounds}")
+            else:
+                # Bounds may not be required but can be optional
+                if self.required_bounds_optional:
+                    print(f"Bounds required: {self.required_bounds} but is optional")
+                else:
+                    print(f"Bounds required: {self.required_bounds} and is not optional")
+                    
+        if hasattr(self, "required_initial_guess") and hasattr(self, "required_initial_guess_optional"):
+            if self.required_initial_guess:
+                print(f"Initial guess required: {self.required_initial_guess}")
+            else:
+                # Bounds may not be required but can be optional
+                if self.required_initial_guess_optional:
+                    print(f"Initial guess required: {self.required_initial_guess} but is optional")
+                else:
+                    print(f"Initial guess required: {self.required_initial_guess} and is not optional")
+
+    def osipi_accepted_dimensions(self):
+        """The array of accepted dimensions
+        e.g.
+        (1D,   2D,   3D,    4D,    5D,    6D)
+        (True, True, False, False, False, False)
+        """
+        
+        return (False,) * 6
+
+    def osipi_accepts_dimension(self, dim):
+        """Query if the selection dimension is fittable"""
+        
+        accepted = self.accepted_dimensions()
+        if dim < 0 or dim > len(accepted):
```

Comment:
I think there's an array offset here in the two getters. The general getter, `osipi_accepted_dimensions` seems to indicate that the array starts at 1D and goes to 6D. But the specific getter, `osipi_accepts_dimension` doesn't have the offset, so `accepted[0]` looks up the first value, which should reference the 1D case.
Perhaps the accepted dimensions should be stored as a dictionary with the dimension as the key and `True` or `False` as the payload?

Reply:
All of the checks are still very much WIP and are untested, non-paid-attention-to draft code :) I prioritized getting the testing functionality up for the abstract. But this is next on the list of things to do. If I remember correctly, no checks are being run in any of the non-check methods, unless the user calls them. Unless you see something being ran, I hope we can disregard them from this PR. I could also do one better and remove them from the current branch. What do you think?

Reply:
Sounds good. I'd say keep them in because it's something we want. Maybe a halfway point would be to comment out the current implementation and always return `True`. That way there's an indication of the goal without the specific implementation details fixed.

Reply:
I'll do that then!